File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p02589/p02589_num4_parsed.py
from collections import defaultdict

def count_string_pairs(N, strings):
    # Create a dictionary to store the frequency count of the first two characters of each string
    freq_count = defaultdict(int)
    
    # Iterate through each string
    for string in strings:
        # Check if the string has at least two characters
        if len(string) >= 2:
            # Increment the frequency count of the first two characters
            freq_count[string[:2]] += 1
    
    # Calculate the number of pairs where Limak can obtain one string from the other
    count = 0
    for freq in freq_count.values():
        count += freq * (freq - 1) // 2
    
    return count


# Read the input
N = int(input())
strings = []
for _ in range(N):
    string = input()
    strings.append(string)

# Call the function and print the output
print(count_string_pairs(N, strings))

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p02589/p02589_num4_parsed.cpp
#include <iostream>
#include <vector>
#include <unordered_set>

using namespace std;

// Function to count the number of unordered pairs
int countUnorderedPairs(vector<string>& strings) {
    // Create a set to store the unique substrings
    unordered_set<string> substrings;
    
    // Iterate through each string
    for (const auto& s : strings) {
        // Remove the first character of the string
        string substring = s.substr(1);
        
        // Check if the substring is already in the set
        if (substrings.count(substring) > 0) {
            // If yes, return 1 as Limak can obtain one string from the other
            return 1;
        }
        
        // Insert the substring into the set
        substrings.insert(substring);
    }
    
    // If no unordered pair found, return 0
    return 0;
}

int main() {
    // Read the value of N
    int N;
    cin >> N;
    
    // Create a vector to store the strings
    vector<string> strings(N);
    
    // Read the strings
    for (int i = 0; i < N; i++) {
        cin >> strings[i];
    }
    
    // Count the number of unordered pairs
    int pairs = countUnorderedPairs(strings);
    
    // Print the result
    cout << pairs << endl;
    
    return 0;
}